src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -63,6 +63,32 @@ public static void ReadClassWithComplexObjects()
             obj.Verify();
         }
 
+        [Fact]
+        public static void ReadClassWithTrailingWhitespace()
+        {
+            ClassWithComplexObjects obj = JsonSerializer.Parse<ClassWithComplexObjects>(ClassWithComplexObjects.s_json + " \r\n\t");
+            obj.Verify();
+            string reserialized = JsonSerializer.ToString(obj);
+
+            // Properties in the exported json will be in the order that they were reflected, doing a quick check to see that
+            // we end up with the same length (i.e. same amount of data) to start.
+            Assert.Equal(ClassWithComplexObjects.s_json.StripWhitespace().Length, reserialized.Length);
+
+            // Shoving it back through the parser should validate round tripping.
+            obj = JsonSerializer.Parse<ClassWithComplexObjects>(reserialized);
+            obj.Verify();
+        }
+
+        [Fact]
+        public static void ReadClassWithTrailingComments()
```

Comment:
Can we add a test with leading comments too?

Reply:
I've consolidated the tests and added both leading and trailing trivia.